src/zmalloc.c

```diff
@@ -28,12 +28,20 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "fmacros.h"
+#include "config.h"
+#include "solarisfixes.h"
```

Comment:
Is there a particular reason for moving these private headers first? Over time it tends to be trouble, so if it must be done I think we should have a single designated `config.h` for that.

Reply:
we copied that block from `server.h` since we needed this part of `fmacros.h`
```c
#if defined(__linux__)
#define _GNU_SOURCE
#define _DEFAULT_SOURCE
#endif
```
without it `madvise` didn't wanna play ball...

in server.h this pile of compatibility headers are included first, so we thought we can copy that entire block here too.

@yossigo what do you suggest? moving `fmacros.h` and `solarisfixes.h` to be part of `config.h`?